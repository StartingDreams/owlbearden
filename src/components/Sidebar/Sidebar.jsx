import React from 'react';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';
import FontIcon from 'material-ui/FontIcon';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, withRouter } from 'react-router-dom';

export class Sidebar extends React.PureComponent {
  render() {
    const { leftDrawerOpen } = this.props;
    return (
      <Drawer open={leftDrawerOpen} containerStyle={{ top: '64px' }} >
        <MenuItem
          leftIcon={<FontIcon className="material-icons">dashboard</FontIcon>}
          containerElement={<NavLink to="/dashboard" />}
          onTouchTap={this.closeLeftDrawerClick}
        >
          Dashboard
        </MenuItem>
        <MenuItem
          leftIcon={<FontIcon className="material-icons">whatshot</FontIcon>}
          containerElement={<NavLink to="/casting/spells" />}
          onTouchTap={this.closeLeftDrawerClick}
        >
          Spells
        </MenuItem>
      </Drawer>
    );
  }
}

Sidebar.propTypes = {
  leftDrawerOpen: PropTypes.bool,
};

export const mapStateToProps = state => ({
  leftDrawerOpen: state.ui.get('leftDrawerOpen'),
});

export default withRouter(connect(
  mapStateToProps,
)(Sidebar));
