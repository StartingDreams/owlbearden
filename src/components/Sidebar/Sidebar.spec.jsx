import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { fromJS } from 'immutable';
import { Sidebar, mapStateToProps } from './Sidebar';

describe('<Sidebar />', () => {
  it('should match the snapshot', () => {
    const leftDrawerOpen = false;
    const wrapper = shallow(<Sidebar leftDrawerOpen={leftDrawerOpen} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('mapStateToProps - leftDrawerOpen should be passed in', () => {
    const leftDrawerOpen = false;
    const ui = fromJS({
      leftDrawerOpen,
    });
    const state = {
      ui,
    };
    const props = mapStateToProps(state);
    expect(props.leftDrawerOpen).toBe(ui.get('leftDrawerOpen'));
  });
});
