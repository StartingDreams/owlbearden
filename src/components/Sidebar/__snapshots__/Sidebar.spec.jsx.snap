// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should match the snapshot 1`] = `
<Drawer
  containerStyle={
    Object {
      "top": "64px",
    }
  }
  disableSwipeToOpen={false}
  docked={true}
  open={false}
  openSecondary={false}
  swipeAreaWidth={30}
  width={null}
  zDepth={2}
>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    containerElement={
      <NavLink
        activeClassName="active"
        to="/dashboard"
      />
    }
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    leftIcon={
      <FontIcon
        className="material-icons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        dashboard
      </FontIcon>
    }
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  >
    Dashboard
  </MenuItem>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    containerElement={
      <NavLink
        activeClassName="active"
        to="/casting/spells"
      />
    }
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    leftIcon={
      <FontIcon
        className="material-icons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        whatshot
      </FontIcon>
    }
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  >
    Spells
  </MenuItem>
</Drawer>
`;
